sherpa/instrument.py

```diff
@@ -413,102 +430,114 @@ def _create_tail_grid(axis_list):
 class PSFModel(Model):
     """Convolve a model by another model or data set.
 
+    At the moment the code does not distinguish between 1D and 2D data
+    and models.
+
     Notes
     -----
     A number of attributes are displayed as parameters, if set, but
     are not handled as parameters. The attributes are: kernel, size,
-    centre, and origin.
+    center, and origin.
 
     """
 
     def __init__(self, name='psfmodel', kernel=None):
+
+        # store the name without the leading "psfmodel." term that Model adds.
         self._name = name
+
         self._size = None
         self._origin = None
         self._center = None
         self._must_rebin = False
+
         self.radial = Parameter(name, 'radial', 0, 0, 1, hard_min=0,
                                 hard_max=1, alwaysfrozen=True)
         self.norm = Parameter(name, 'norm', 1, 0, 1, hard_min=0, hard_max=1,
                               alwaysfrozen=True)
+
         self.kernel = kernel
         self.model = None
         self.data_space = None
         self.psf_space = None
-        Model.__init__(self, name)
+        super().__init__(name)
 
-    def _get_center(self):
-        if self._center is not None:
-            if len(self._center) == 1:
-                return self._center[0]
-        return self._center
+    def _get_field(self, name):
+        """Return the field value"""
+        field = getattr(self, name)
+        if field is not None and len(field) == 1:
+            return field[0]
+
+        return field
+
+    def _set_field(self, name, vals):
```

Comment:
I know it's a pain to write docs for stuff like this and this is internal, but given that it's all about reformatting, it might be worth it to write down what "vals" is supposed to be is a docstring for the benefit of future developers. Somethink like "vals is any value, scalar or array. If it's not a list, tuple or np.array, this routine will wrap it into a one-element array before setting it. (Why? (I don't know)"

Reply:
I added a note in #1477